### PR TITLE
Fix Deno 1.7.5 --isolatedModules Flag Change – Fixes #28

### LINKED ISSUE #28: denox on deno 1.7.5
## 【Bug】deno throw error

- Bug Report

`
Check https://denopkg.com/BentoumiTech/denox/denox.ts
error: TS1205 [ERROR]: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
export { ParseOptions, parse, parseAll } from "./yaml/parse.ts";
         ~~~~~~~~~~~~
    at https://deno.land/std@0.42.0/encoding/yaml.ts:6:10

TS1205 [ERROR]: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
  DumpOptions as StringifyOptions,
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.42.0/encoding/yaml.ts:8:3

TS1205 [ERROR]: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
  DenoWorkspace,
  ~~~~~~~~~~~~~
    at https://raw.githubusercontent.com/BentoumiTech/denox/master/src/interfaces.ts:25:3

TS1205 [ERROR]: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
  WorkspaceGlobal,
  ~~~~~~~~~~~~~~~
    at https://raw.githubusercontent.com/BentoumiTech/denox/master/src/interfaces.ts:26:3

TS1205 [ERROR]: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
  WorkspaceScript,
  ~~~~~~~~~~~~~~~
    at https://raw.githubusercontent.com/BentoumiTech/denox/master/src/interfaces.ts:27:3

TS1205 [ERROR]: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
  WorkspaceOptions,
  ~~~~~~~~~~~~~~~~
    at https://raw.githubusercontent.com/BentoumiTech/denox/master/src/interfaces.ts:28:3

TS1205 [ERROR]: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
  DenoOptionsEntries,
  ~~~~~~~~~~~~~~~~~~
    at https://raw.githubusercontent.com/BentoumiTech/denox/master/src/interfaces.ts:29:3

TS1205 [ERROR]: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
  DenoOptionValue,
  ~~~~~~~~~~~~~~~
    at https://raw.githubusercontent.com/BentoumiTech/denox/master/src/interfaces.ts:30:3

TS1205 [ERROR]: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
export { getOptionType, OptionTypeValues };
                        ~~~~~~~~~~~~~~~~
    at https://raw.githubusercontent.com/BentoumiTech/denox/master/src/deno_options/utils.ts:40:25

TS1205 [ERROR]: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
export { buildDenoCLIOptionsArgs, CLIArgument };
                                  ~~~~~~~~~~~
    at https://raw.githubusercontent.com/BentoumiTech/denox/master/src/deno_options/build_cli_arguments.ts:75:35

Found 10 errors.

`

## Info

- DenoX version: @last version

### PR DESCRIPTION
**Pull request by bluebinary**
## Description

This pull request fixes the recently introduced `deno` incompatibility associated with the Deno project's recent change in handling of the export of types from TypeScript files so that isolated modules can be supported correctly going forward.

As of Deno version 1.5.0, the default value of the `--isolatedModules` flag is now `true`, and as of Deno version 1.7.5, support has been removed for overriding the `--isolatedModules` flag back to `false`, as such it is no longer possible to rely on the older syntax for exporting types under recent and any future versions of `deno`.

Deno as of version 1.5 (unless explicitly overridden via the `--isolatedModules` flag), and since version 1.7.5 (by default, without an option to override), now requires types to be exported using the `export types` syntax, rather than the previously supported and more general `export` syntax.

**Types of changes**

This is a Bugfix change. Running the tests that `denox` supplies, including running `denox` against the `example/` project provided within the repository, indicates that the small syntax changes needed to resolve this issue have not had any impact on the functionality of the `denox` tool.

**Related links**

- https://github.com/BentoumiTech/denox/issues/28
- https://deno.land/manual@v1.7.5/typescript/faqs#why-are-you-forcing-me-to-use-isolated-modules-why-can#39t-i-use-const-enums-with-deno-why-do-i-need-to-do-export-type

#### Fixed

This pull request fixes the incompatibilities between `denox` and `deno` versions 1.7.5 and above, introduced by Deno's removal of support for overriding the `--isolatedModules` flag.

### CODE REVIEW

**Comment by bluebinary** [comment]
Not included in the pull request description above, but included previously in the commit message:

It was also necessary to update `denox` to use a more recent release of the `deno` standard library YAML encoder/parser, upgrading to the most recent release (as of writing) under `std@0.88.0` vs `std@0.42.0`, due to the `deno` YAML encoder’s own use of the general `export` syntax for exports under the earlier `std@0.42.0` version.

**Review by fangmarks — APPROVED** [review]

**Review by Cyberhan123 — APPROVED** [review]

**Comment by jsjoeio** [comment]
Any chance this will get merged soon?

**Review by jsjoeio — APPROVED** [review]

**Comment by bluebinary** [comment]
Is anyone with write access to this repository able to approve the pull request? If so, it looks like we will then be able to have the fix merged into the main branch. Thank you all for this very helpful utility.

**Comment by jsjoeio** [comment]
I'll see if I can get in touch with the maintainer on Twitter.

**Comment by jessehanley** [comment]
Hey @jsjoeio and @bluebinary,

Thanks for the Twitter DM @jsjoeio. 

I was working with @BentoumiTech (the maintainer) on my business until recently. Due to some personal issues, he's gone offline. A little bit scary as a few people have been trying to get in touch with him without much luck. He's an amazing, diligent guy so I imagine when/if he comes back this will get sorted.

In the meantime, I'd possibly recommend forking the project and continuing this journey elsewhere.

Let's hope and pray that he's in good health wherever he is. 

Cheers, 
J

**Review by BentoumiTech — APPROVED** [review]
LGTM

**Comment by BentoumiTech** [comment]
Thank you @bluebinary for the pull request, I couldn't look at it before it's very helpful

**Pull request merged**
